### ISSUE TITLE
iPhone X: SnackBar should apply bottom media padding

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#13595]
## Steps to Reproduce

On the iPhone X, snackbars should be inset with additional bottom padding to account for the home screen indicator, as well as symmetric left and right padding to account for the notch when in landscape mode.

![screen shot 2017-12-14 at 18 56 26](https://user-images.githubusercontent.com/351029/34024729-aa9c408a-e100-11e7-8d11-d2439fae9ae6.png)

## Flutter Doctor

```
[✓] Flutter (on Mac OS X 10.13.2 17C89, locale en-AU, channel master)
    • Flutter at /Users/cbracken/src/flutter/flutter
    • Framework revision dc9c95375f (26 minutes ago), 2017-12-14 18:32:41 -0800
    • Engine revision e0d19e2897
    • Tools Dart version 1.25.0-dev.11.0
    • Engine Dart version 2.0.0-edge.cf1de7d46cd88e204380e8f96a993439be56b24c
```

**Comment by cbracken** [comment]
For future archaeologists: in case an iOS devicelab benchmark regression leads back here, it's worth noting that this is the first patch landed after the devicelab machines were updated from Xcode 8 to Xcode 9.

**Comment by alardizabal** [comment]
Hi @cbracken, I think the fix above is causing the issue we see in the Hamilton app. On the main feed where the snackbar is displayed from the bottom of the screen, it looks pretty good. However, on screens where we present a snackbar on top of a persistent footer, it looks overly tall on the iPhone X.

Looks fine when its presented from the bottom
![feed](https://user-images.githubusercontent.com/3614291/34958715-0f5a1d3a-fa01-11e7-9305-ce62bbac3970.png)

Too much padding when when presented on top of a footer view
![detail](https://user-images.githubusercontent.com/3614291/34958717-11efcab8-fa01-11e7-8424-1ccab1815ef3.png)

**Comment by cbracken** [comment]
Thanks for the report. Your hunch sounds exactly right. This should be a relatively quick one to fix. I’ll take a look.

**Comment by cbracken** [comment]
Bottom padding in the presence of a  bottom navigation bar or persistent footer buttons fixed by #14396.
